designer/server/src/plugins/routes/api.js

```diff
@@ -1,30 +1,31 @@
-import { Schema } from '@defra/forms-model'
+import { formDefinitionSchema } from '@defra/forms-model'
+import Joi from 'joi'
 
 import * as forms from '~/src/lib/forms.js'
 
 /**
- * @type {ServerRoute}
+ * @type {ServerRoute<{ Params: FormByIdInput }>}
  */
 export const getFormWithId = {
   method: 'GET',
   path: '/api/{id}/data',
   options: {
-    async handler(request, h) {
-      const { id } = request.params
-      try {
-        const formJson = await forms.getDraftFormDefinition(id)
-
-        return h.response(/** @type {object} */ (formJson))
-      } catch (error) {
-        request.logger.error(['GET /api/{id}/data', 'getFormWithId'], error)
-        throw error
-      }
+    /**
+     * @param {FormByIdInput} request
+     */
```

Comment:
Ah sorry, is just a `{ id: string }` type

You'd need `import('@hapi/hapi').Request` with it as a generic:

```suggestion
    /**
     * @param {Request<{ Params: FormByIdInput }>} request
     */
```

<br>

But because we've typed it already at the route level (above) don't need it here at all 🙌

```suggestion
```